### DISCUSSION TITLE
Feature Request: allow disabling all animations and popups

### DISCUSSION THREAD

**Discussion by sionescu** [ghostty-org/ghostty#3242] [Feature Requests, Ideas] [answered]
I'd like to be able to disable all transition animations and the popup messages that show when a command is executed (like "Copied to clipboard").

**Comment by iZarrios** [comment]
I think there were only 2 pop-ups/animations(maybe there is more, I am not sure yet) and one of them can be disabled [resize-overlay](https://ghostty.org/docs/config/reference#resize-overlay) but as you noted the "Copied to clipboard" toast does not have a config flag to be disabled, the change seems to be pretty minor (I tried doing it myself and I will be using my patch until this gets to the main Ghostty build).
In case someone is interested [here](https://github.com/iZarrios/ghostty/commit/100954cb6c2d7464255b6883f259b9966fb1f042) is what I had to change.

**Comment by sionescu** [comment]
Opening and closing a tab produces a transition animation in the tab bar, which I'd like to eliminate.
As for your patch, I suppose it looks good but I'd like a single toggle to eliminate all "toast" messages.

**Comment by gpanders** [comment]
An option to only show "Copied to clipboard" popup when an application copies via OSC 52 would be good to have too. The popup may not be as useful when the user explicitly copies with e.g. Ctrl+Shift+C, but is much more useful when an application is doing it (a badly behaved application could be copying to the clipboard without the user knowing).

**Comment by mitchellh** [comment]
This feels like a request for upstream (libadw/GTK). These are the frameworks doing the animation. Is there even an API or way for us to disable them? cc @tristan957

**Reply by jcollie** [reply]
The "Copied to clipboard" popup is something we trigger ourselves. We just need a config entry to enable/disable that behavior.
 https://github.com/ghostty-org/ghostty/blob/94599102e9fb8247af08cbbbcb7ee25e3d31e1bd/src/apprt/gtk/Window.zig#L898

**Reply by mitchellh** [reply]
Yes I know that but I'm talking about the _animations_ around it. But more importantly around new tab and so on pointed out by OP.

**Reply by jcollie** [reply]
Ah, I found:

https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/method.Animation.set_follow_enable_animations_setting.html

and 

https://docs.gtk.org/gtk4/property.Settings.gtk-enable-animations.html

so in theory if you have animations globally disabled we should not be showing any animations, but that should be tested.

**Reply by tristan957** [reply, marked as answer]
https://docs.gtk.org/gtk4/property.Settings.gtk-enable-animations.html

People can set this in their GTK config files

**Reply by mitchellh** [reply]
This is the answer. That's what I'd expect and follows macOS conventions as well.

**Reply by iZarrios** [reply]
What about an option to disable the "Copied to clipboard" toast? I personally find it annoying or maybe something as @gpanders suggested?

**Reply by jcollie** [reply]
> What about an option to disable the "Copied to clipboard" toast? I personally find it annoying or maybe something as @gpanders suggested?

That will require an new configuration option.

**Reply by iZarrios** [reply]
I am aware, what's wrong with that?

**Reply by tristan957** [reply]
We could have an `in-app-notifications` config option with values like `all,none,no-copy,no-config-reload`. Let's see what Mitchell says in terms of designs. I'm happy to add that support and accommodate this.

**Reply by mitchellh** [reply]
I already merged `adw-toast` for this reason: https://github.com/ghostty-org/ghostty/commit/8c74b807047c47eaac9b005cdd7828a05ce44f43

It may make sense to make that not Adw or Linux-specific but there aren't currently any notifications for macOS.

**Reply by iZarrios** [reply]
Oh nice!

**Reply by bebosudo** [reply]
For anyone reading this, to disable the `Copied on clipboard` spam see https://github.com/ghostty-org/ghostty/issues/4800#issuecomment-2685774252
